api/access/server.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package dps
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+
+	"github.com/optakt/flow-dps/models/index"
+)
+
+// Server is a simple implementation of the generated AccessAPIServer interface.
+// It uses an index reader interface as the backend to retrieve the desired data.
+// This is generally an on-disk interface, but could be a GRPC-based index as
+// well, in which case there is a double redirection.
+type Server struct {
+	index index.Reader
+	codec index.Codec
+}
+
+// NewServer creates a new server, using the provided index reader as a backend
+// for data retrieval.
+func NewServer(index index.Reader, codec index.Codec) *Server {
+
+	s := Server{
+		index: index,
+		codec: codec,
+	}
+
+	return &s
+}
+
+func (s *Server) Ping(ctx context.Context, in *access.PingRequest) (*access.PingResponse, error) {
+	panic("implement me")
```

Comment:
This goes throughout this PR, but let's assume somebody actually uses this code before everything is implemented.

- Do you think we want to crash the server when using unsupported functionality, even though the rest of the features might already be working?
- Do you think it's a clear error message that makes it obvious why it crashed, or errored, when that function was called?

I prefer returning an error here that says at least `"not implemented"`, because it addresses those two points.

Reply:
Sure, no probs. I just left the Goland implementation that is generated when using the `Implement Interface` feature, didn't think much of it since we will implement all of the methods before it is ever used 😄